packages/compiler/src/render3/view/template.ts

```diff
@@ -1327,192 +1323,7 @@ export class TemplateDefinitionBuilder implements t.Visitor<void>, LocalResolver
-          () => this.convertPropertyBinding(value));
-    }
-
-    // Note that we generate an implicit `on idle` if the `deferred` block has no triggers.
-    // `deferOnIdle()`
-    if (idle || (!prefetch && Object.keys(triggers).length === 0)) {
-      this.creationInstruction(
-          idle?.sourceSpan || null, prefetch ? R3.deferPrefetchOnIdle : R3.deferOnIdle);
-    }
-
-    // `deferOnImmediate()`
-    if (immediate) {
-      this.creationInstruction(
-          immediate.sourceSpan, prefetch ? R3.deferPrefetchOnImmediate : R3.deferOnImmediate);
-    }
-
-    // `deferOnTimer(1337)`
-    if (timer) {
-      this.creationInstruction(
-          timer.sourceSpan, prefetch ? R3.deferPrefetchOnTimer : R3.deferOnTimer,
-          [o.literal(timer.delay)]);
-    }
-
-    // `deferOnHover(index, walkUpTimes)`
-    if (hover) {
-      this.domNodeBasedTrigger(
-          'hover', hover, metadata, prefetch ? R3.deferPrefetchOnHover : R3.deferOnHover);
-    }
-
-    // `deferOnInteraction(index, walkUpTimes)`
-    if (interaction) {
-      this.domNodeBasedTrigger(
-          'interaction', interaction, metadata,
-          prefetch ? R3.deferPrefetchOnInteraction : R3.deferOnInteraction);
-    }
-
-    // `deferOnViewport(index, walkUpTimes)`
-    if (viewport) {
-      this.domNodeBasedTrigger(
-          'viewport', viewport, metadata,
-          prefetch ? R3.deferPrefetchOnViewport : R3.deferOnViewport);
-    }
-  }
-
-  private domNodeBasedTrigger(
-      name: string,
-      trigger: t.InteractionDeferredTrigger|t.HoverDeferredTrigger|t.ViewportDeferredTrigger,
-      metadata: R3DeferBlockMetadata, instructionRef: o.ExternalReference) {
-    const triggerEl = metadata.triggerElements.get(trigger);
-
-    // Don't generate anything if a trigger cannot be resolved.
-    // We'll have template diagnostics to surface these to users.
-    if (!triggerEl) {
-      return;
-    }
-
-    this.creationInstruction(trigger.sourceSpan, instructionRef, () => {
-      const location = this.elementLocations.get(triggerEl);
-
-      if (!location) {
-        throw new Error(
-            `Could not determine location of reference passed into ` +
-            `'${name}' trigger. Template may not have been fully analyzed.`);
-      }
-
-      // A negative depth means that the trigger is inside the placeholder.
-      // Cap it at -1 since we only care whether or not it's negative.
-      const depth = Math.max(this.level - location.level, -1);
-      const params = [o.literal(location.index)];
-
-      // The most common case should be a trigger within the view so we can omit a depth of
-      // zero. For triggers in parent views and in the placeholder we need to pass it in.
-      if (depth !== 0) {
-        params.push(o.literal(depth));
-      }
-
-      return params;
-    });
+    throw new Error('Deferred blocks no longer supported in TemplateDefinitionBuilder');
```

Comment:
Context here is that for the `TemplateDefinitionBuilder` to work, we would have to resolve the triggers during analysis. We already have the logic, but given that this isn't necessary in the template pipeline and that the pipeline is the default, I figured it was a good time to clean up the logic.

Reply:
I'd propose adding more info into the error message to make sure that it's not interpreted as "defer is no longer supported", here is a proposal:
```suggestion
    throw new Error('Deferred blocks are no longer supported by the obsolete TemplateDefinitionBuilder, ' +
      'Please enable the new template compilation pipeline for your application.');
```